lib/logthrsource/logthrsourcedrv.c

```diff
@@ -216,30 +212,38 @@ log_threaded_source_worker_free(LogPipe *s)
   log_source_free(s);
 }
 
-static LogThreadedSourceWorker *
-log_threaded_source_worker_new(GlobalConfig *cfg)
+void
+log_threaded_source_worker_init_instance(LogThreadedSourceWorker *self, LogThreadedSourceDriver *driver,
+                                         gint worker_index)
 {
-  LogThreadedSourceWorker *self = g_new0(LogThreadedSourceWorker, 1);
-  log_source_init_instance(&self->super, cfg);
+  log_source_init_instance(&self->super, log_pipe_get_config(&driver->super.super.super));
   main_loop_threaded_worker_init(&self->thread, MLW_THREADED_INPUT_WORKER, self);
-  self->thread.thread_init = log_threaded_source_worker_thread_init;
-  self->thread.thread_deinit = log_threaded_source_worker_thread_deinit;
-  self->thread.run = log_threaded_source_worker_run;
-  self->thread.request_exit = log_threaded_source_worker_request_exit;
+  self->thread.thread_init = _worker_thread_init;
+  self->thread.thread_deinit = _worker_thread_deinit;
+  self->thread.run = _worker_thread_run;
+  self->thread.request_exit = _worker_thread_request_exit;
 
   wakeup_cond_init(&self->wakeup_cond);
 
-  self->super.super.init = log_threaded_source_worker_init;
+  self->super.super.init = _worker_init;
   self->super.super.free_fn = log_threaded_source_worker_free;
-  self->super.wakeup = _worker_wakeup;
```

Comment:
the wakeup method assignment is needed here. kira failed for this reason.

If I add it back it works correctly.

Reply:
:facepalm: Well, this was accidental :) Thanks for checking it out!